### ISSUE TITLE
"All entities" option missing from Home Assistant OS but present in Docker

### ISSUE THREAD

**Issue by HomeAssistantPro** [home-assistant/frontend#17962]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

In my Home Assistant **OS** 2023.9.2 installation, when I create a light group under Settings > Devices & services > Helpers > Create Helper > Group > Light Group, I am able to add my lights as expected. 

See screenshot #1:
![#1 - HA OS version](https://github.com/home-assistant/frontend/assets/145419758/bc01376f-2d8e-40f4-9937-9b434f1540da)

When I edit the group to toggle "All entities" however, that option is missing from the UI. 

See screenshot #2 and #3:
![#2 - Edit light group](https://github.com/home-assistant/frontend/assets/145419758/429a1064-67c7-49d1-a529-5a3e41e7d38c)

![#3 - all entities missing](https://github.com/home-assistant/frontend/assets/145419758/15978685-0ecf-4b6e-95da-e9e95679df49)

### Describe the behavior you expected

In my Home Assistant **Docker** 2023.9.2 installation however, the "All entities" option is present and acts as expected. This is what I expect to see also in the OS installation.

See screenshots #4, #5, and #6.

Tl;dr, it appears the "All entities" group helper option exists on Docker but not on Home Assistant OS?

![#4 HA Docker version](https://github.com/home-assistant/frontend/assets/145419758/a48c8928-c0a2-4c7d-a699-d06d62db6860)

![#5 edit light group on Docker](https://github.com/home-assistant/frontend/assets/145419758/40b2cbf2-ede8-414a-b445-d85fcfe01fab)

![#6 - all entities exist on Docker](https://github.com/home-assistant/frontend/assets/145419758/c62b2c6b-d5e9-4875-906b-170a4fc5bec2)

### Steps to reproduce the issue

1. Log into a Home Assistant OS 2023.9.2 installation (I installed via ESXi .ova file)
2. Navigate to:  Settings > Devices & services > Helpers > Create Helper > Group > Light Group
3. Name group and add lights (mine were these [wifi TP-Link KL135P2 bulbs from Amazon](https://www.amazon.com/dp/B0964DN9TV?ref=ppx_yo2ov_dt_b_product_details&th=1))
4. Edit group
5. Choose "Group options"
6. All entities toggle missing.

### What version of Home Assistant Core has the issue?

2023.9.2

### In which browser are you experiencing the issue with?

Mozilla Firefox 117.0.1

### Which operating system are you using to run this browser?

Windows 11 22H2

### Additional information

I tried a private window in Firefox and also tried Chrome and Edge, but the problem persisted, even after wiped caches.

I downgraded my **Docker** installation to 2023.8.4 and it worked there also.

My OS installation was via the ESXi .ova file. FWIW, I spun up a few VMs of older Home Assistant OS installs on my ESXi host, and the problem existed all the way back to at least **10.0**. I didn't go past that.

Every version listed in Settings > About in my current Home Assistant OS install:

- Home Assistant 2023.9.2
- Supervisor 2023.09.2
- Operating System 10.5
- Frontend 2023.98.11.0 - latest

**Comment by karwosts** [comment]
I believe this setting is hidden behind "advanced mode", can you check if you have that enabled in both installations?

**Comment by HomeAssistantPro** [comment]
> I believe this setting is hidden behind "advanced mode", can you check if you have that enabled in both installations?

Ugh! Thanks for pointing that out, I didn't know the Advanced Setting affected this option.
